lib/availability_resolver.js

```diff
@@ -93,67 +93,51 @@ class AvailabilityResolver {
-            if (recapCustomerCodes[item.recapCustomerCode]) item.eddRequestable = recapCustomerCodes[item.recapCustomerCode[0]].eddRequestable
-
-            // Make sure properties exist:
-            if (!item.status || item.status.length === 0) item.status = [{}]
-            if (!item.reqestable || item.requestable.length === 0) item.requestable = []
-            if (!item.physRequestable || item.physRequestable.length === 0) item.physRequestable = []
-
+          if (recapAvailabilityStatus !== "Item Barcode doesn't exist in SCSB database.") {
             // NYPL items require checking 1) recap status, location requestability, and sierra status requestability
             if (isItemNyplOwned(item)) {
               // First check recap status:
               if (recapAvailabilityStatus === 'Available') {
                 item.status[0].id = config.get('itemAvailability.available.id')
                 item.status[0].label = config.get('itemAvailability.available.label')
                 // Now that true availability is set, establish requestability based on it (and holdingLocation)
-                item.requestable[0] = requestableBasedOnStatusAndHoldingLocation(item)
-                item.physRequestable[0] = requestableBasedOnStatusAndHoldingLocation(item)
+                item.physRequestable = requestableBasedOnStatusAndHoldingLocation(item)
+                // check eddRequestable by recap customer code:
+                if (recapCustomerCodes[item.recapCustomerCode]) item.eddRequestable = recapCustomerCodes[item.recapCustomerCode[0]].eddRequestable
               } else {
                 // If recap status is Not Available, we need check nothing else
-                item.requestable[0] = false
-                item.physRequestable[0] = false
-                item.eddRequestable = false
                 item.status[0].id = config.get('itemAvailability.notAvailable.id')
                 item.status[0].label = config.get('itemAvailability.notAvailable.label')
               }
 
-            // It's a partner item in ReCAP:
+              // It's a partner item in ReCAP:
             } else {
               if (recapAvailabilityStatus === 'Available') {
-                item.requestable[0] = true
-                item.physRequestable[0] = true
+                item.physRequestable = true
+                if (recapCustomerCodes[item.recapCustomerCode]) item.eddRequestable = recapCustomerCodes[item.recapCustomerCode[0]].eddRequestable
                 item.status[0].id = config.get('itemAvailability.available.id')
                 item.status[0].label = config.get('itemAvailability.available.label')
               } else {
-                item.requestable[0] = false
-                item.physRequestable[0] = false
                 item.status[0].id = config.get('itemAvailability.notAvailable.id')
                 item.status[0].label = config.get('itemAvailability.notAvailable.label')
```

Comment:
It looks like there are more opportunities for light refactoring here. Looks like one of the things this function is doing is applying `item.status` values. What `status` value it sets depends on `recapAvailabilityStatus`. The code does this twice depending on `isItemNyplOwned(item)`, but actually it sets it in the same way regardless of item ownership. So that part could be moved upward in scope. Looks like `item.eddRequestable` is also currently set in two places based on item ownership, but is evaluated identically, so could be moved up. The only check that truly depends on item ownership that I see is `item.physRequestable` (because we just set it to `true` automatically for partner records that are Available.) Am I reading this right? So we could make this once sprawling method even shorter.